plugins/scaffolder-backend/src/scaffolder/stages/prepare/github.ts

```diff
@@ -20,22 +20,46 @@ import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
 import { parseLocationAnnotation } from '../helpers';
 import { InputError, Git } from '@backstage/backend-common';
 import { PreparerBase, PreparerOptions } from './types';
+import { Logger } from 'winston';
 import GitUriParser from 'git-url-parse';
+import { Config } from '@backstage/config';
+import {
+  GitHubIntegrationConfig,
+  readGitHubIntegrationConfigs,
+} from '@backstage/integration';
 
 export class GithubPreparer implements PreparerBase {
-  token?: string;
+  private readonly integrations: GitHubIntegrationConfig[];
+  private readonly scaffolderToken: string | undefined;
+  private readonly logger: Logger;
 
-  constructor(params: { token?: string } = {}) {
-    this.token = params.token;
+  constructor(config: Config, { logger }: { logger: Logger }) {
+    this.logger = logger;
+    this.integrations = readGitHubIntegrationConfigs(
+      config.getOptionalConfigArray('integrations.github') ?? [],
+    );
+
+    if (!this.integrations.length) {
+      this.logger.warn(
+        'Integrations for Github in Scaffolder are not set. This will cause errors in a future release. Please migrate to using integrations config and specifying tokens under hostnames',
```

Comment:
> Please migrate to using integrations config and specifying tokens under hostnames

Is it possible to improve this message by giving out an example? Either print in the console
```
integrations:
  github:
...
```

or a permalink to a few lines of an example config file? That would help the migrations a bit.

Reply:
Use GitHub with that capitalization in all "regular" text